prompt_poet/template.py

```diff
@@ -84,18 +110,24 @@ def logger(self) -> str:
 
     @property
     def template_name(self) -> str:
-        """The name of the template file."""
+        """The name of the template file.
+
+        Deprecated: This property will be removed in a future release.
```

Comment:
is this true? i think it would be useful, no? 

you should at least add 

```
    @property
    def template_id(self) -> str:
```

given that you have that on the TemplateLoader...and in any case it might not be a bad idea to keep `template_name` and `template_dir` as pass through props and populate them on the TemplateLoader..

Reply:
yeah, i can add template_id. But there are cases there are no template_dir or template_name anymore. For example, you can write a template_loader support loading from feature_flag, or cache.

Reply:
added template_id.

Reply:
Makes sense; I think template_id seems sufficient